test/ruby_lsp_rails/server_test.rb

```diff
@@ -268,6 +268,38 @@ def print_it!
     $> = original_stdout
   end
 
+  test "forked processes are named based on caller" do
+    skip("Fork is not supported on Windows") if Gem.win_platform?
+
+    File.write("my_addon.rb", <<~RUBY)
+      class MyServerAddon < RubyLsp::Rails::ServerAddon
+        def name
+          "MyAddon"
+        end
+
+        def execute(request, params)
+          pid = fork do
+            # We can't directly send a message in theses tests because we're using a StringIO as stdout instead of the
+            # actual pipe, which means that the child process doesn't have access to the same object
+            File.write("process_name.txt", $0)
+          end
+
+          Process.wait(pid)
+          send_message({ process_name: File.read("process_name.txt") })
+          File.delete("process_name.txt")
```

Comment:
Let's make sure this gets cleaned?

```suggestion
        ensure
          File.delete("process_name.txt")
```